R/data_rename.R

```diff
@@ -82,6 +82,21 @@ data_rename <- function(data,
     replacement <- paste0(seq_along(pattern))
   }
 
+  # coerce to character
+  replacement <- as.character(replacement)
+
+  # check if `replacement` has no empty strings and no NA values
+  invalid_replacement <- is.na(replacement) | !nzchar(replacement)
+  if (any(invalid_replacement)) {
+    insight::format_error(
+      "`replacement` is not allowed to have `NA` or empty strings.",
```

Comment:
Sorry for being nitpicky but this error is a bit confusing if someone passes `pattern = c(test = "Species", "Sepal.Length")` (we didn't provide `replacement` so it shouldn't appear in the message). When we check that `pattern` has names above, can we throw an error like "Either name all elements of `pattern` or use `replacement`" if some elements don't have any names?